Add enableSelectionMode for DialogList

zipangulu opened the pull request:
It would be very useful to add also to DialogList the functionality "enableSelectionMode" so that you can delete and perform other actions on the dialogs

mradzinski committed:
Allow the user to get the current position of a DIALOG

This should close issue #29. I believe that providing the user with the whole list of DIALOG objects is a bit "dangerous", users can modify that collection and feed it back to the adapter and given the current implementation that's not something we might want. So having a ```getDialogPosition()``` method should solve the case when the user needs to have the current adapter position of a certain dialog.

troy379 committed:
Update README.md

ynagarjuna2012 committed:
Update COMPONENT_DIALOGS_LIST.MD

Just small change here 

`* adapter.addItem(DIALOG dialog) - adds one dialog to the end of the list
* adapter.addItem(int position, DIALOG dialog) -  adds a new dialog to the specified position.`

toanpv committed:
Add TypingListener to MessageInput
Add sample

toanpv committed:
Remove spaces

toanpv committed:
Add delay typing to attr
Add TypingListener to docs

bevzaanton committed:
Merge pull request #56 from stfalcon-studio/v.0.2.1

V.0.2.1

bevzaanton committed:
DialogList: fixed changing text style to normal

bevzaanton committed:
MessageList new feature: registering click listener for view by id

bevzaanton committed:
removed some unnecessary sample code

bevzaanton committed:
Merge pull request #57 from stfalcon-studio/v0.2.2

V0.2.2

bevzaanton committed:
Update README.md

bevzaanton committed:
Update COMPONENT_MESSAGES_LIST.md

troy379 committed:
Update README.md

Commit:
tiny refactoring

hbmartin committed:
Added upsert(Message) method to add or update message to adapter as appropriate
Useful when connecting adapter directly to streaming data source
Changed update methods to return boolean

hbmartin committed:
remove application tag from manifest

bevzaanton committed:
Merge pull request #39 from ynagarjuna1995/patch-1

Update COMPONENT_DIALOGS_LIST.MD

kirvis250 committed:
npe check in DialofsListAdapter.java for last message when exists none

philippeauriach committed:
added get item by id method

philippeauriach committed:
allow moving an item from position x to position y

philippeauriach committed:
test jitpack compile

philippeauriach committed:
Revert "test jitpack compile"

This reverts commit f2e5cc7ce237130799b365ddd705448fe07b877e.

philippeauriach committed:
please codebeat

SeunAdelekan committed:
Made correction to DialogList documentation

A typographic error was made in the documentation for the DialogList component under the 'Prepare your model' section.
"IDialog" was written as "IDealog".
This pull request corrects that error.

dmathewwws committed:
Add nullable for getImageUrl()

Add the option for nullable to distinguish between a message and image. Useful for when using Kotlin.

alouanemed committed:
fix link in docs

Tibolte committed:
wording

bevzaanton committed:
Merge pull request #140 from Tibolte/master

wording

Sulfkain committed:
Avoid Crash on empty list

If some empty list passed to insert at the end, will crash with a indexOutOfBounds.

To avoid this, it's recomended to add a check to see if the list is empty or not. And adding a @NotNull anotation to the param will be useful to use it with Kotlin.

If you don't like a "fast" return, do it with a tipical if.

mario commented:
Hello,

considering that the library is not maintained and I actively use it, I have decided to fork it for the time being under the same name. Changes to naming and other practices will come in the future.

In the mean time, I will look at your PR and consider merging it.

I hope we can revive this project together.

https://github.com/mario/chatkit

Cheers,
Mario

andriizhumela committed:
update support library version

andriizhumela committed:
update gradle version version

andriizhumela committed:
remove old command from gradle

andriizhumela committed:
remove old tags from gradle

anymol committed:
Merge pull request #1 from anymol/library_version

Upgrade Library and Gradle version

alexz89ua committed:
Merge pull request #164 from anymol/master

Upgrade Library and Gradle version

bevzaanton committed:
Fixed artifacts on some devices in ShapeImageView. Changed fixture message image.

bevzaanton committed:
Merge pull request #146 from Sulfkain/patch-2

Avoid Crash on empty list in addToEnd

bevzaanton committed:
Merge pull request #134 from alouanemed/patch-1

Fix link typo in docs

bevzaanton committed:
Merge pull request #119 from dmathewwws/dmathewwws-patch-1

Add nullable for getImageUrl()

bevzaanton committed:
Merge pull request #112 from SeunAdelekan/master

Made correction to DialogList documentation

bevzaanton committed:
Merge pull request #70 from philippeauriach/allow-moving-item

Allow moving Dialog item and get Dialog by id

bevzaanton committed:
Merge pull request #32 from mradzinski/patch-2

Allow the user to get the current position of a DIALOG

bevzaanton committed:
Version 0.2.3

bevzaanton committed:
hot fix

bevzaanton committed:
Update README.md

bevzaanton committed:
Fixed Add attachments button in default view demo

bevzaanton committed:
Merge remote-tracking branch 'github/master'

bevzaanton committed:
Create LICENSE

bevzaanton committed:
Merge pull request #167 from stfalcon-studio/add-license-1

Create LICENSE

bevzaanton committed:
Merge pull request #61 from hbmartin/master

Added upsert(Message) method to add or update message to adapter as appropriate

bevzaanton committed:
Merge pull request #75 from kirvis250/npe-fix

npe check in DialofsListAdapter.java when there is no last message (is null)

bevzaanton committed:
Merge branch 'add_typing_listener' of https://github.com/toanpv/ChatKit into toanpv-add_typing_listener

# Conflicts:
#	docs/STYLES_ATTR.md
#	sample/src/main/java/com/stfalcon/chatkit/sample/features/demo/def/DefaultMessagesActivity.java

bevzaanton committed:
Merge branch 'add_typing_listener' of https://github.com/toanpv/ChatKit into toanpv-add_typing_listener

# Conflicts:
#	docs/STYLES_ATTR.md
#	sample/src/main/java/com/stfalcon/chatkit/sample/features/demo/def/DefaultMessagesActivity.java

bevzaanton committed:
Merge branch 'toanpv-add_typing_listener'

bevzaanton committed:
Update LICENSE

bevzaanton committed:
Update README.md

bevzaanton committed:
Fixed artifacts with bubble background in message list.
DialogList: reverseLayout=false for default

bevzaanton committed:
Merge remote-tracking branch 'github/master'

bevzaanton committed:
MessageListAdapter: added notifyDataSetChanged to method "clear"

bevzaanton committed:
Fixed bug "The totalItemsCount parameter in OnLoadMore callback contains date header." Issue #86
Added separate method MessagesListAdapter.clear(notifyDataSetChanged). Method MessagesListAdapter.clear() does notifyDataSetChanged by default. Issue #89

bevzaanton committed:
v0.3.0

bevzaanton committed:
Update README.md

bevzaanton committed:
Update CHANGELOG.md

joostfunkekupper committed:
Removal of forced software layer rendering

bevzaanton committed:
Merge pull request #178 from joostfunkekupper/master

Removal of forced software layer rendering

bevzaanton committed:
add ability to passing custom data to ViewHolder. Version 0.3.1

bevzaanton committed:
Update documentation

bevzaanton committed:
set deprecated ViewHolder constructor. Update sample.

bevzaanton committed:
Merge remote-tracking branch 'github/master'

bevzaanton committed:
Update README.md

bevzaanton committed:
add proguard rules

bevzaanton committed:
Merge remote-tracking branch 'github/master'

bevzaanton committed:
Update CHANGELOG.md

bevzaanton committed:
Update README.md

bevzaanton committed:
fix in proguard rules

bevzaanton committed:
Merge remote-tracking branch 'github/master'

mweathers committed:
Enable payloads for custom content types

mathroule committed:
Clear last message if null in dialog list (Fix issue #189)

mathroule committed:
Add upsert method to dialogs list

bevzaanton committed:
Merge pull request #191 from mathroule/upsert-dialog

Add upsert method to dialogs list

bevzaanton committed:
Merge pull request #190 from mathroule/clear-last-message-if-null-in-dialog-list

Clear last message if null in dialog list (Fix issue #189)

bevzaanton committed:
ver 0.3.2

bevzaanton committed:
fix addItem method in DialogListAdapter

bevzaanton committed:
Merge branch 'master' of https://github.com/stfalcon-studio/ChatKit into v0.3.2

bevzaanton committed:
make protected items field in MessagesListAdapter and DialogsListAdapter

bevzaanton committed:
fix delete and deleteByIds methods in MessagesListAdapter

bevzaanton committed:
ability to pass custom data in ImageLoader

bevzaanton committed:
Merge pull request #185 from mweathers/master

Enable payloads for custom content types

bevzaanton committed:
Merge branch 'master' of https://github.com/stfalcon-studio/ChatKit into v0.3.2

mathroule committed:
Update documentation to add upsertItem method

bevzaanton committed:
Merge pull request #208 from mathroule/add-upsert-to-documentation

Update documentation to add upsertItem method

bevzaanton committed:
Merge branch 'master' of https://github.com/stfalcon-studio/ChatKit into v0.3.2

bevzaanton committed:
ability to pass custom data in ImageLoader

Commit:
Added upsert method to move a specific item to the start.

Commit:
Changed access modifier from public to protected for:
- List<Wrapper> items
- method `generateDateHeaders`

bevzaanton committed:
Update CHANGELOG.md

bevzaanton committed:
Update CHANGELOG.md

bevzaanton committed:
Merge pull request #209 from PrimozPernat/move_to_start_if_update

Added upsert method to move a specific item to the start.

bevzaanton committed:
version 0.3.3-SNAPSHOT

bevzaanton committed:
version 0.3.3

bevzaanton committed:
Update CHANGELOG.md

bevzaanton committed:
Update README.md

bevzaanton committed:
Update COMPONENT_MESSAGES_LIST.md

andriizhumela committed:
Update README.md

andriizhumela committed:
Update COMPONENT_MESSAGES_LIST.md

IlyaGazman committed:
Migrated to AndroidX and Java 8

muankit committed:
Updated broken link 

Link for [custom layout looks like](https://github.com/stfalcon-studio/ChatKit/blob/master/sample/src/main/res/layout/item_custom_dialog.xml) at line 224 is broken.

bevzaanton committed:
Merge pull request #338 from gazman-sdk/master

Migrated to AndroidX and Java 8

bevzaanton committed:
update Picasso version

bevzaanton committed:
update gradle-wrapper.properties

bevzaanton committed:
add gradle-wrapper.jar

bevzaanton committed:
v0.4.1

bevzaanton committed:
Update README.md

bevzaanton committed:
Update README.md

bevzaanton committed:
Update README.md

bevzaanton committed:
Update README.md

bevzaanton committed:
Merge pull request #345 from muankit/patch-1

Updated broken link
